subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/notes/tasks/UpdateReleaseNotes.java

```diff
@@ -42,25 +43,33 @@ void updateReleaseNotes(boolean previewMode, File releaseNotesFile, String newCo
         }
     }
 
-    //TODO SF deduplicate and unit test
     static Map<String, Contributor> contributorsMap(Collection<String> contributorsFromConfiguration,
                                                     ProjectContributorsSet contributorsFromGitHub,
-                                                    Collection<String> developers) {
+                                                    Collection<String> developers,
+                                                    String githubUrl) {
         Map<String, Contributor> out = new HashMap<String, Contributor>();
-        for (String contributor : contributorsFromConfiguration) {
+        out.putAll(transform(contributorsFromConfiguration, githubUrl));
+        out.putAll(transform(contributorsFromGitHub.getAllContributors()));
+        out.putAll(transform(developers, githubUrl));
+        return out;
+    }
+
+    private static Map<String, Contributor> transform(Collection<String> contributors, String githubUrl) {
```

Comment:
Since the two transform methods in this class have different behaviour, let's name them accordingly. It would be easier to read this code without looking for the type of variables to see which method is used. Maybe sth like "transformToDefaultContributorsByName" and "groupByName"?

Reply:
I already have plans to do some further refactoring in this area and most likely this part of the code will change again.
We could rename the methods but I think it does not add a lot of value.
@wwilk, are you ok with merging it as it is and I will come up with a larger refactoring during the next days?

Reply:
Yeah I guess removing hardcoded value is more important at the moment. Looking forward to this refactoring!